### PR TITLE
[Observability] Per-adapter L1<->L2 throughput metrics

### PR DESCRIPTION
**Pull request by Oasis-Git**
**What this PR does / why we need it**:

**Special notes for your reviewers**:

**If applicable**:

- [ ] this PR contains user facing changes - docs added
- [ ] this PR contains unit tests

---

> [!NOTE]
> **Medium Risk**
> Adds new event types/metadata and wires new metrics subscribers into hot-path store/prefetch controllers; mistakes could skew telemetry or increase overhead but should not affect cache correctness.
> 
> **Overview**
> Adds per-adapter L1↔L2 observability by emitting richer L2 controller events (`task_id`, `l2_name`, `total_bytes`) and introducing new `L2_LOAD_TASK_*` events for per-adapter load attribution.
> 
> Registers a new `L2ThroughputSubscriber` that correlates submit→complete pairs to export sampled throughput histograms (`lmcache_mp.l2_{store,load}_throughput_gbs`) labeled by `l2_name`, and extends existing L2 counters to support `l2_name`-labeled completion counters (plus a new `lmcache_mp.l2_load_completed`) for dashboard-derived per-backend IOPS.
> 
> Updates docs/contracts and adds unit tests validating labeled counters and the new throughput histogram behavior (sampling, correlation, edge cases, and EventBus integration).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 00969dae80bf706d8a19b9d11654d01096f1bdf2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: lmcache/v1/distributed/storage_controllers/store_controller.py [hunk 1]
```diff
@@ -170,6 +170,11 @@ class InFlightStoreTask:
     """The subset of keys for which reserve_read succeeded
     (i.e., keys holding an L1 read lock that must be released)."""
 
+    total_bytes: int = 0
+    """Total bytes submitted for this task, carried forward so the
+    L2_STORE_COMPLETED event can report the same value as SUBMITTED
+    without re-reading MemoryObjs after their locks are released."""
+
```
Comment (ApostaC) [inline lmcache/v1/distributed/storage_controllers/store_controller.py]: Is this required? IIUC, we could emit the total size for the `L2_STORE_SUBMITTED` event, and match the submit event + get back the bytes when the `L2_STORE_COMPLETED` is received. 

In this case, we don't need to introduce a new field in `total_bytes`

File: lmcache/v1/distributed/storage_controllers/prefetch_controller.py [hunk 2]
```diff
@@ -154,6 +154,10 @@ class InFlightPrefetchRequest:
     load_plan: dict[int, Bitmap] = field(default_factory=dict)
     # Load phase: adapter_idx -> task_id (removed as results arrive)
     pending_load_tasks: dict[int, L2TaskId] = field(default_factory=dict)
+    # Load phase: adapter_idx -> total bytes submitted to that adapter.
+    # Carried across submit->complete so the L2_LOAD_TASK_COMPLETED event
+    # can report the same value as SUBMITTED without re-reading MemoryObjs.
+    pending_load_bytes: dict[int, int] = field(default_factory=dict)
```
Comment (ApostaC) [inline lmcache/v1/distributed/storage_controllers/prefetch_controller.py]: Similar to store operations, we can emit the total bytes when the task is submitted, and get the number back when receiving the load task finish event -- so that we don't need to carry over the pending_load_bytes.

**Review by ApostaC — COMMENTED** [review]
Otherwise LGTM

**Review by ApostaC — APPROVED** [review]
LGTM!

**Review by deng451e — APPROVED** [review]
